products/BellHybrid/apps/common/src/widgets/TimeSetSpinnerVertical.cpp

```diff
@@ -148,44 +202,69 @@ namespace gui
         const auto t = std::localtime(&time);
 
         if (timeFormat == Locale::TimeFormat::FormatTime24H) {
-            hour->set_value(t->tm_hour);
-            minute->set_value(t->tm_min);
+            const auto hours   = std::chrono::hours{t->tm_hour};
+            const auto time24H = date::make24(hours, isPM());
+            auto stringHour = std::to_string(time24H.count());
+            if(t->tm_hour < doubleDigit) {
+                hour_first->setText("0");
+                hour_second->setText(stringHour.substr(0,1));
+            }
+            else {
+                hour_first->setText(stringHour.substr(0,1));
+                hour_second->setText(stringHour.substr(1,1));
+            }
+
+            auto stringMin = std::to_string(t->tm_min);
+            if(t->tm_min < doubleDigit) {
+                minute_first->setText("0");
+                minute_second->setText(stringMin.substr(0,1));
+            }
+            else {
+                minute_first->setText(stringMin.substr(0,1));
+                minute_second->setText(stringMin.substr(1,1));
+            }
         }
         else {
             const auto hours   = std::chrono::hours{t->tm_hour};
             const auto time12H = date::make12(hours);
-            hour->set_value(time12H.count());
-            minute->set_value(t->tm_min);
-        }
+            auto stringHour = std::to_string(time12H.count());
 
-        handleContentChanged();
-    }
+            if(t->tm_hour < doubleDigit) {
+                hour_first->setText("0");
+                hour_second->setText(stringHour.substr(0,1));
+            }
+            else {
+                hour_first->setText(stringHour.substr(0,1));
+                hour_second->setText(stringHour.substr(1,1));
+            }
 
-    auto TimeSetSpinnerVertical::getTime() const noexcept -> std::time_t
-    {
-        using namespace utils::time;
-        const auto now     = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
-        const auto newTime = std::localtime(&now);
-        if (timeFormat == Locale::TimeFormat::FormatTime24H) {
-            newTime->tm_hour = hour->value();
-        }
-        else {
-            newTime->tm_hour = date::make24(std::chrono::hours{hour->value()}, isPM()).count();
+            auto stringMin = std::to_string(t->tm_min);
+             if(t->tm_min < doubleDigit) {
+                minute_first->setText("0");
+                minute_second->setText(stringMin.substr(0,1));
+            }
+            else {
+                minute_first->setText(stringMin.substr(0,1));
+                minute_second->setText(stringMin.substr(1,1));
+            }
```

Comment:
This operation is being done multiple times in this code. Maybe we can create helper function for that and reuse it? I belive the code readability would increase